Pcap++/header/PcapFilter.h

```diff
@@ -243,62 +243,145 @@ namespace pcpp
 	class IPFilter : public IFilterWithDirection
 	{
 	private:
-		std::string m_Address;
+		IPAddress m_Address;
 		std::string m_IPv4Mask;
```

Comment:
btw, we have `IPNetwork`, maybe we should use it?

Reply:
yes, I agree - we can probably refactor some of the c'tors to accept `IPNetwork` and use it instead of a string

Reply:
Perhaps we can do a slightly larger refactor of this class and use `IPNetwork` to represent masks - it already supports IPv4/IPv6 netmasks so should have a lot of the functionality that currently exists in the filter class

Reply:
It can be done with `IPNetwork` but its a bigger change. 

The setters `setAddr`, `setMask` and `setLen` would also have to be replaced with `setNetwork`/`getNetwork`. The `IPFilter` would still need to keep the data set from the deprecated setters for the deprecation period as the setters can't create a new `IPNetwork` object by themselves to update the filter.

Parsing to a filter string would be simpler tho, as the new filter would generally always use the `net <netAddr>/<prefixLen>` syntax, instead of mixing between it and `net <netAddr> mask <maskAddr>` syntax.

Reply:
Since we already have many breaking changes, perhaps it's okay to deprecate functions, and even remove the function if necessary. Need to confirm with @seladb.

Reply:
@seladb Seems good.

Reply:
@seladb @tigercosmos 
Ok, refactored `IPFilter` to use `IPNetwork`. Did not deprecate any of the old setters although switching IPv6 to IPv4 address essentially gives a network of 1 address if the prefix length is over 32 as it caps it at 32.

General overview of the changes include:
- `IPNetwork` (+ v4 and v6) getting a new explicit 1 parameter constructor that makes a 1 address network.
- `IPv4Network` and `IPv6Network` having their exception messages about `invalid netmask` expanded.
- The `PCPP_DEPRECATED` macro being moved from 'HttpLayer.h' to 'DeprecationUtils.h'. I thought I might use it, but I ended up not needing it.
- All of `IPFilter`'s methods now work for IPv4 and IPv6 (`setMask` is no longer IPv4 exclusive, yay). Netmasks still need to be matched IPv4 address with IPv4 netmask format, and vice versa for IPv6.